### ISSUE TITLE
[Bug]: Character count is not announced to screen readers

### ISSUE THREAD

**Issue by cjrace** [dfe-analytical-services/shinyGovstyle#193]
### Overview

Character count isn't announced to screen reader users.

### Steps to Reproduce

Inspect the HTML of the showcase app

```
<div class="govuk-hint govuk-character-count__message">
 <span>You have used</span>
 <span id="text_area2wc">0</span>
 <span id="text_area2wl">of the 300 allowed</span>
</div>
```

### Expected Behavior

Screen reader comments:

> “While navigating the page with NVDA and entering content into the text area input field just before the ‘check for errors button,’ I found that on using the ‘arrow keys’ to explore the page identifies a character count which is not indicated for users of screen reader software while typing. I found this confusing, as it is easy to miss this information as it appears after the input field, and if tabbing to the next control is not accessed. Including a clear indication to the character count by announcing this after the first few words typed, or if a pause in typing has occurred for example, will resolve this issue for users of screen reading software.”

### Additional Information

Can review against the GOV.UK character count component - https://design-system.service.gov.uk/components/character-count/.

Page 40 of the September 2025 accessibility audit report.
